### ISSUE TITLE
`skaffold init --analyze` builder ordering

### ISSUE THREAD

**Issue by matthewmichihara** [GoogleContainerTools/skaffold#3863]
In Java projects such as https://github.com/GoogleCloudPlatform/cloud-code-samples/tree/master/java/java-hello-world, there is a `Dockerfile` and a `pom.xml` with Jib configured.

```
$ skaffold init --XXenableJibInit --analyze --skip-deploy | jq
{
  "builders": [
    {
      "name": "Docker",
      "payload": {
        "path": "Dockerfile"
      }
    },
    {
      "name": "Jib Maven Plugin",
      "payload": {
        "path": "pom.xml",
        "project": "org.springframework.boot:hello-world"
      }
    }
  ]
}
```
Cloud Code IntelliJ uses the ordering of the builders returned here to determine the default builder. If Docker and Jib builders are detected, can Jib take precedence?

Related: https://github.com/GoogleContainerTools/skaffold/issues/3661#issuecomment-592732189

**Comment by chanseokoh** [comment]
@TadCordle

This is because file entries in a directory is [sorted](https://github.com/GoogleContainerTools/skaffold/blob/86b290e83c43934200f73e11532af3cf01c2308f/pkg/skaffold/initializer/analyze/analyze.go#L97-L99) for deterministic search behavior.

I think this can be re-ordered on the Skaffold side. However, I wonder if Clode Code IntelliJ should go over the list anyway, because there may be multiple projects (and sub-projects) to resolve?

**Comment by balopat** [comment]
It would be great to know 
 1) the typical reasons for having multiple builders defined in the same directory, that build the exact same context.
 2) what the IDE is doing with the default builder? 


1)**typical reasons for having multiple builders** -  In this case: is a sample, we have both builders for educational reasons - however I could see maybe some complex organizations allow/enforce using different builders in different contexts (i.e. Dockerfile has the flexibility to add some extra flavor that is mandated in prod, but during dev people prefer jib as it's faster??) - but in general this should be rare - so maybe I don't care that much which one is chosen by default? 
 2) **what the IDE is doing with the default builder?** is this just a temporary solution for the K8s flows before we introduce builder selection? In that case I think it is also just a temporary choice, that takes off the pressure of "getting it right forever". I'd be more okay with choosing jib - as it is more deterministic in terms of the settings to build a jib project. Dockerfile's can be found easily but the rest of the settings (context, target, etc.) are hard to determine automatically, while a well configured jib plugin can contain all that's needed. 

We can represent the precedence in sorting if that helps.

**Comment by matthewmichihara** [comment]
> It would be great to know
> 
> 1. the typical reasons for having multiple builders defined in the same directory, that build the exact same context.

Yeah hm.. I don't know. We have this situation in the cloud-code-samples sample projects, but those are mainly educational.

> 2. what the IDE is doing with the default builder?

This popped up in the context of Cloud Run, where the run configuration builder combobox defaults to the first builder that is returned by `skaffold init --analyze`, and for Java, it was returning Docker before Jib.

But yes the issue also exists for the auto-generated Skaffold run configs for Kubernetes projects, and I think even when we have a way to select a detected builder here, the IDEs would want to default to the first builder returned by Skaffold.

We can also try to perform some ordering in the IDE, but it seems like it would make the most sense to do this in Skaffold.
